scripts/irods/test/test_rule_engine_plugin_framework.py

```diff
@@ -202,6 +202,82 @@ def test_continuation_followed_by_an_error(self):
+            config.server_config['log_level']['rule_engine'] = 'trace'
+            config.server_config['log_level']['legacy'] = 'trace'
+            config.server_config['plugin_configuration']['rule_engines'].insert(0, {
+                'instance_name': 'irods_rule_engine_plugin-passthrough-instance',
+                'plugin_name': 'irods_rule_engine_plugin-passthrough',
+                'plugin_specific_configuration': {
+                    'return_codes_for_peps': [
+                        {
+                            'regex': '^' + pep_api_exec_my_rule_pre + '$',
+                            'code': RULE_ENGINE_CONTINUE
+                        },
+                        {
+                            'regex': '^' + pep_api_exec_rule_expression_pre + '$',
+                            'code': RULE_ENGINE_CONTINUE
+                        }
+                    ]
+                }
+            })
+            lib.update_json_file_from_dict(config.server_config_path, config.server_config)
+
+            IrodsController().restart(test_mode=True)
+
+            # Test 1 - exec_rule_text
+            # ~~~~~~~~~~~~~~~~~~~~~~~
+            # Capture the current size of the log file. This will be used as the starting
+            # point for searching the log file for a particular string.
+            log_offset = lib.get_file_size_by_path(paths.server_log_path())
+
+            msg = 'exec_rule_text: Using -r to target a REP is not required anymore!'
+            self.admin.assert_icommand(['irule', 'writeLine("stdout", "{0}")'.format(msg), 'null', 'ruleExecOut'], 'STDOUT', msg)
+
+            with open(paths.server_log_path(), 'r') as log_file:
+                msg = "Returned '{0}' to REPF.".format(str(RULE_ENGINE_CONTINUE))
+                mm = mmap.mmap(log_file.fileno(), 0, access=mmap.ACCESS_READ)
+                found = (mm.find(msg, log_offset) != -1)
+                mm.close()
+                self.assertTrue(found)
+
+            # Test 2 - exec_rule_expression
+            # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+            log_offset = lib.get_file_size_by_path(paths.server_log_path())
+
+            msg_2 = 'exec_rule_expression: Using -r to target a REP is not required anymore!'
+            self.admin.assert_icommand(['irule', 'delay("0.1s") {{ writeLine("serverLog", "{0}"); }}'.format(msg_2), 'null', 'null'])
+
+            def check_log():
+                with open(paths.server_log_path(), 'r') as log_file:
+                    msg_1 = "Returned '{0}' to REPF.".format(str(RULE_ENGINE_CONTINUE))
+                    mm = mmap.mmap(log_file.fileno(), 0, access=mmap.ACCESS_READ)
+                    index = mm.find(msg_1, log_offset)
+                    found = (index != -1 and mm.find(msg_2, index) != -1)
+                    mm.close()
+                    return found
+
+            delayAssert(check_log)
+
+        IrodsController().restart(test_mode=True)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unexpected keyword argument 'test_mode' in method call](https://app.codacy.com/manual/trel/irods/pullRequest?prid=5009095)